lib/ansible/plugins/strategy/__init__.py

```diff
@@ -395,35 +395,35 @@ def parent_handler_match(target_handler, handler_name):
                                         listening_handler = search_handler_blocks(listening_handler_name, iterator._play.handlers)
-                                        if listening_handler is None:
-                                            raise AnsibleError("The requested handler listener '%s' was not found in any of the known handlers" % listening_handler_name)
+
                                         if original_host not in self._notified_handlers[listening_handler]:
                                             self._notified_handlers[listening_handler].append(original_host)
-                                            display.vv("NOTIFIED HANDLER %s" % (listening_handler_name,))
+                                            display.vv("NOTIFIED LISTENED HANDLER %s" % (listening_handler_name,))
+                                            handler_found = True
 
-                                else:
-                                    target_handler = search_handler_blocks(handler_name, iterator._play.handlers)
-                                    if target_handler is not None:
-                                        if original_host not in self._notified_handlers[target_handler]:
-                                            self._notified_handlers[target_handler].append(original_host)
-                                            # FIXME: should this be a callback?
-                                            display.vv("NOTIFIED HANDLER %s" % (handler_name,))
-                                    else:
-                                        # As there may be more than one handler with the notified name as the
-                                        # parent, so we just keep track of whether or not we found one at all
-                                        found = False
-                                        for target_handler in self._notified_handlers:
-                                            if parent_handler_match(target_handler, handler_name):
-                                                self._notified_handlers[target_handler].append(original_host)
-                                                display.vv("NOTIFIED HANDLER %s" % (target_handler.get_name(),))
-                                                found = True
-
-                                        # and if none were found, then we raise an error
-                                        if not found:
-                                            raise AnsibleError("The requested handler '%s' was found in neither the main handlers list nor the listening handlers list" % handler_name)
+                                # and if none were found, then we raise a warning
+                                if not handler_found:
+                                    display.warning("The requested handler '%s' was found in neither the main handlers list nor the listening handlers list" % handler_name)
```

Comment:
This should be an error by default, created #17902 to allow this to be configurable so this change should be removed from here.